timestamp-client/src/main/java/com/palantir/timestamp/CorroboratingTimestampService.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * (c) Copyright 2018 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.palantir.timestamp;
+
+import javax.annotation.concurrent.ThreadSafe;
+
+/**
+ * A timestamp service decorator for introducing runtime validity checks on received timestamps.
+ */
+@ThreadSafe
+public class CorroboratingTimestampService implements TimestampService {
+
+    private final TimestampService delegate;
```

Comment:
Could we just put this inside the actual impl? It does feel auxiliary like tracing/stats collection, since it's critical to correctness. Plus it does mean less wrapper/delegate things.

Reply:
(I think we mean _doesn't_ feel auxiliary right?)

The impl is, at the root, a Feign proxy so we'll need to delegate at some point - though we could see if there's a way to consolidate things